mitiq/zne/inference.py

```diff
@@ -397,12 +378,183 @@ def is_converged(self) -> bool:
             )
         return len(self._outstack) == len(self._scale_factors)
 
+    def _batch_populate_instack(self) -> None:
+        """Populates the instack with all parameters at which to compute
+        expectation values.
+        """
+        if self._shot_list:
+            self._instack = [
+                {"scale_factor": scale, "shots": shots}
+                for scale, shots in zip(self._scale_factors, self._shot_list)
+            ]
+        else:
+            self._instack = [
+                {"scale_factor": scale} for scale in self._scale_factors
+            ]
+
+    def run_batched(
+        self,
+        qp: QPROGRAM,
+        batched_executor: Callable[..., List[float]],
+        scale_noise: Callable[[QPROGRAM, float], QPROGRAM],
+        num_to_average: int = 1,
+    ) -> "BatchedFactory":
+        """Computes the expectation values at each scale factor by calling
+        `batched_executor` on a list of all circuits to run.
+
+        Args:
+            qp: Quantum circuit to run.
+            batched_executor: Function which inputs a list of circuits and
+                outputs a list of expectation values.
+            scale_noise: Noise scaling function.
+            scale_noise: Noise scaling function.
+            num_to_average: Number of times to call scale_noise at each scale
+                factor.
+        """
+        self.reset()
+        self._batch_populate_instack()
+        kwargs = self._get_keyword_args()
```

Comment:
Code cleanliness comment -- looks like this method is only used by the `BatchedFactory`, and as such could be moved to this class definition.

Reply:
Moved `_get_keyword_args` to `BatchedFactory` instead of `BaseFactory`.